File: src/0754/S0754.c
int reachNumber(int target){
    if (target < 0) target = -target;
    int n = sqrt(target<<1) + 0.5;
    int s = n * n + n >> 1;
    return (s == target || (s - target & 1) == 0) ? n : n + 1 + (n&1);
}

File: src/0754/S0754.rs
impl Solution {
    pub fn reach_number(target: i32) -> i32 {
        let mut t = if target < 0 {-target} else {target};
        let n = (((t<<1) as f64).sqrt() + 0.5) as i32;
        let s = n * n + n >> 1;
        if (s - t) & 1 == 0 || s == t {n} else {n + 1 + (n&1)} 
    }
}
